wp-includes/class-wp-web-app-manifest.php

```diff
@@ -59,9 +59,22 @@ public function init() {
 	 * Mainly copied from Jetpack_PWA_Manifest::render_manifest_link().
 	 */
 	public function manifest_link_and_meta() {
+		$manifest = $this->get_manifest();
 		?>
 		<link rel="manifest" href="<?php echo esc_url( rest_url( self::REST_NAMESPACE . self::REST_ROUTE ) ); ?>">
-		<meta name="theme-color" content="<?php echo esc_attr( $this->get_theme_color() ); ?>">
+		<meta name="theme-color" content="<?php echo esc_attr( $manifest['theme_color'] ); ?>">
+		<meta name="apple-mobile-web-app-capable" content="yes" />
+		<meta name="mobile-web-app-capable" content="yes" />
+		<meta name="apple-touch-fullscreen" content="YES" />
+		<?php if ( ! empty( $manifest['icons'] ) ) : ?>
+			<?php if ( ! empty( $manifest['icons'][1] ) && ! empty( $manifest['icons'][1]['src'] ) ) : ?>
+				<link rel="apple-touch-startup-image" href="<?php echo esc_url_raw( $manifest['icons'][1]['src'] ); ?>">
+			<?php elseif ( ! empty( $manifest['icons'][0] ) && ! empty( $manifest['icons'][0]['src'] ) ) : ?>
+				<link rel="apple-touch-startup-image" href="<?php echo esc_url_raw( $manifest['icons'][0]['src'] ); ?>">
```

Comment:
Is there a need for duplication here? To me it seems what you want to do is sort all icons by size and then get the first one, yes?

For example, first given this change:

```diff
diff --git a/wp-includes/class-wp-web-app-manifest.php b/wp-includes/class-wp-web-app-manifest.php
index b7c9552..21f77fd 100644
--- a/wp-includes/class-wp-web-app-manifest.php
+++ b/wp-includes/class-wp-web-app-manifest.php
@@ -181,12 +181,12 @@ class WP_Web_App_Manifest {
 	 *
 	 * Mainly copied from Jetpack_PWA_Manifest::build_icon_object() and Jetpack_PWA_Helpers::site_icon_url().
 	 *
-	 * @return array|null $icon_object An array of icons, or null if there's no site icon.
+	 * @return array $icon_object An array of icons, which may be empty.
 	 */
 	public function get_icons() {
 		$site_icon_id = get_option( 'site_icon' );
 		if ( ! $site_icon_id || ! function_exists( 'get_site_icon_url' ) ) {
-			return null;
+			return array();
 		}
 
 		$icons     = array();
```

Then change the above code to:


```php
$manifest = $this->get_manifest();
$icons = $manifest['icons'];
usort(
	$icons,
	function( $a, $b ) {
		return intval( strtok( $a['sizes'], 'x' ) ) - intval( strtok( $b['sizes'], 'x' ) );
	}
);
$icon = array_shift( $icons );
if ( ! empty( $icon ) ) ?>
	...Now print the link...
<?php endif;
```